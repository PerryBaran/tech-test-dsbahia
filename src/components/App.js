import React, { useState } from "react";
import Search from "./Search";
import "../styles/app.css";
import SearchResults from "./SearchResults";

function App() {
  const [searchResults, setSearchResults] = useState([]);
  console.log('searchResults:', searchResults);
  return (
    <div className="app">
      <img className="nasa-logo" src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg" alt="Nasa Logo" />
      <Search setSearchResults={setSearchResults} />
      <SearchResults />
    </div>
  );
}

export default App;
