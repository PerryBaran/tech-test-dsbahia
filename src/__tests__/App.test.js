import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("Renders Correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Displays NASA Logo", () => {
    render(<App />);
    const logoElement = screen.getByAltText("Nasa Logo");
    expect(logoElement).toBeInTheDocument();
    expect(logoElement.src).toContain("https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg");
  });

  it("Displays Search Component", () => {
    render(<App />);
    const searchElement = screen.getByTestId("search-component");
    expect(searchElement).toBeInTheDocument();
  });
});
