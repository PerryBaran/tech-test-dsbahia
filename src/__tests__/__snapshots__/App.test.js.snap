// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <img
      alt="Nasa Logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      data-testid="search-component"
    >
      <form>
        <input
          class="search"
          placeholder="Enter keywords..."
          type="text"
          value=""
        />
        <button
          class="search-button"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
